src/lib/http/proxy-agent.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * External dependencies
+ */
+import { SocksProxyAgent } from 'socks-proxy-agent';
+import { HttpsProxyAgent } from 'https-proxy-agent';
+import { getProxyForUrl } from 'proxy-from-env';
+import debug from 'debug';
+
+/**
+ * Internal dependencies
+ */
+
+// Note: This module requires the use of a special environment variable "VIP_PROXY_OTHER_ENABLED"
+// The setting of it to any value allows this module to create a proxy agent based on proxy environment variables
+// If not set, this module will revert back to the previous functionality (hence being fully backward compatible and non-breaking for users)
+
+// This function returns a proxy given a few scenarios (in order of precedence):
+// 1. VIP_PROXY is set: a SOCKS proxy is returned same as the previous version of this module
+// 2. No applicable variables are set: null is returned (thus, no proxy agent is returned)
+// 3. VIP_PROXY_OTHER_ENABLED and HTTPS_PROXY are set: an HTTPS_PROXY is returned (assuming the given url uses the https protocol)
+// 4. NO_PROXY is set along with VIP_PROXY_OTHER ENABLED and HTTPS_PROXY: An HTTPS_PROXY is returned assuming NO_PROXY is not applicable to the URL
+// Note: HTTP_PROXY is not supported at this time as the wp url is always https in production
+function createProxyAgent( url ) {
+	const HTTPS_PROXY = process.env.HTTPS_PROXY || process.env.https_proxy || null;
+	const VIP_PROXY = process.env.VIP_PROXY || process.env.vip_proxy || null;
+	const NO_PROXY = process.env.NO_PROXY || process.env.no_proxy || null;
+
+	// VIP Socks Proxy should take precedence, should be fully backward compatible
+	if ( VIP_PROXY ) {
+		debug( `Enabling VIP_PROXY proxy support using config: ${ VIP_PROXY }` );
+		return new SocksProxyAgent( VIP_PROXY );
+	} else if ( process.env.VIP_PROXY_OTHER_ENABLED && ! coveredInNoProxy( url, NO_PROXY ) && HTTPS_PROXY ) {
```

Comment:
@frank-cerny I've spent some time thinking about this, here's my take, let me know your thoughts.

The whole goal of this PR is to enable people using the CLI to use the Systems proxy they have configured.
So, if we're not using the `VIP_PROXY` we might find ourselves using the `HTTPS_PROXY` environment variable, or even the `HTTP_PROXY` one .

It also seems that some apps allow `HTTP_PROXY` with https sites. The reason why this is possible is that in those cases we are using a proxy that works in `HTTP`. So the communication with the proxy will be unencrypted.

Therefore, after some thought, I was thinking this:
Instead of having a more generic `VIP_PROXY_OTHER_ENABLED` what about changing it to `VIP_USE_SYSTEM_PROXY` and then return the first we find between
* `SOCKS_PROXY`
* `HTTPS_PROXY`
* `HTTP_PROXY`

Always keeping in mind the `NO_PROXY` config.

This would allow us to remove the check on the protocol and also to make full use of whatever proxy configuration we might have on the final system, while, at the same time, allowing us to disable the proxy if we need it. 

What do you think? 
It could look something like
```js
if (process.env.VIP_PROXY_OTHER_ENABLED && ! coveredInNoProxy( url, NO_PROXY )) {
  if ( SOCKS_PROXY ) {
	  debug( `Enabling SOCKS proxy support using config: ${ HTTPS_PROXY }` );
	  return new SocksProxyAgent( SOCKS_PROXY );
  }
  if ( HTTPS_PROXY ) {
	  debug( `Enabling HTTPS proxy support using config: ${ HTTPS_PROXY }` );
	  return new HttpsProxyAgent( HTTPS_PROXY );
  }
  if ( HTTP_PROXY ) {
	  debug( `Enabling HTTP proxy support using config: ${ HTTP_PROXY }` );
	  return new HttpProxyAgent( HTTP_PROXY );
  }
}

Reply:
@andrea-sdl I think this is a solid idea. I've used https -> http proxies so your scenario makes sense to me. This should simplify the implementation even more too (as your snippet shows).

I also like how there is still a "feature flag" (VIP_USE_SYSTEM_PROXY) to ensure its all backwards compatible. VIP_USE_SYSTEM_PROXY sounds WAY better than VIP_PROXY_OTHER_ENABLED too haha.

I think as long as the "order of precedence" of proxies is documented (socks -> https -> http) then this approach will work great!'

I can make these updates (i.e. "borrow" your snippet and update the tests within the next couple of days. Thank you for the feedback! (Keep it coming :) )

Reply:
@andrea-sdl Changes have been made. Now I fully see the allure of table driven tests. Please let me know if I missed anything or you have any additional feedback!